test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/DotnetCommand.cs

```diff
@@ -15,6 +16,35 @@ public DotnetCommand()
         public DotnetCommand(string dotnetUnderTest)
             : base(dotnetUnderTest)
         {
+            // Some scripts (e.g. RunCsc/RunVbc) depend on the correct `dotnet`
+            // being in the PATH. When tests use `DotnetUnderTest.WithBackwardsCompatibleRuntimes`,
+            // this means that the up-to-date dotnet command is resolved by the
+            // old-runtime-targeting RunCsc/Roslyn, causing compatability issues.
+            // So, detect if we're running a dotnet that's not just a plain "dotnet",
+            // and if so, make sure it's on the PATH.
```

Comment:
I think we should arrange in the product without mucking with PATH that a build initiated by a given `dotnet` uses the same `dotnet` to launch the compiler that ships with the CLI. Perhaps we can set something other than PATH that RunCsc/RunVbc can use. It can fallback to PATH if that thing is not set.

This feels like we're patching the test logic without fixing a product issue.